Update Runtime Provisioner documentation (#784)
kyma-incubator/compass: docs/provisioner/08-02-provisioning-gardener.md

  4. Download the service account configuration (`kubeconfig.yaml`) and use it to create a Secret in the `compass-system` Namespace with the key named `credentials` and the value encoded with base64.

     ```bash
     kubectl -n compass-system create secret generic {SECRET_NAME} --from-file=credentials={PATH_TO_KUBECONFIG}
     ```
    

>   5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on GCP. Note, that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory, and that the order of the components matters.

Reviewer: ```suggestion
  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on GCP. Note that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory and that the order of the components matters.
```

                providerSpecificConfig: { gcpConfig: { zone: "europe-west4-a" } }
              }
            }
            kymaConfig: {
              version: "1.8.0"
              components: [
                { component: "compass-runtime-agent", namespace: "compass-system" }
                {
                  component: "{KYMA_COMPONENT_NAME}"
                  namespace: "{NAMESPACE_TO_INSTALL_COMPONENT_TO}"
                  configuration: [
                    { key: "{CONFIG_PROPERTY_KEY}"
                      value: "{CONFIG_PROPERTY_VALUE}"
                      secret: {true|false} # Specifies if the property is confidential
                    }
                  ]
                }
              ]
              configuration: [
                { 
                  key: "{CONFIG_PROPERTY_KEY}"
                  value: "{CONFIG_PROPERTY_VALUE}"
                  secret: {true|false} # Specifies if the property is confidential
                }
              ]
            }

>             credentials: { secretName: "{GARDENER_SERVICE_ACCOUNT_CONFIGURATION_SECERT_NAME}" }

Reviewer 2: We should remove this part as credentials are now passed with envs

>   4. Download the service account configuration (`kubeconfig.yaml`) and use it to create a Secret in the `compass-system` Namespace with the key `credentials` and the value encoded with base64.

Reviewer 2: We should move this part to requirements, as we expect that kubeconfig is already present during Provisioner installation

     ```bash
     kubectl -n compass-system create secret generic {SECRET_NAME} --from-file=credentials={PATH_TO_KUBECONFIG}
     ```
    

>   5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on Azure. Note, that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory, and that the order of the components matters.

Reviewer: ```suggestion
  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on Azure. Note that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory and that the order of the components matters.
```

                autoScalerMax: 4
                maxSurge: 4
                maxUnavailable: 1
                providerSpecificConfig: { azureConfig: { vnetCidr: "10.250.0.0/19" } }
              }
            }
            kymaConfig: {
              version: "1.8.0"
              components: [
                { component: "compass-runtime-agent", namespace: "compass-system" }
                {
                  component: "{KYMA_COMPONENT_NAME}"
                  namespace: "{NAMESPACE_TO_INSTALL_COMPONENT_TO}"
                  configuration: [
                    { key: "{CONFIG_PROPERTY_KEY}"
                      value: "{CONFIG_PROPERTY_VALUE}"
                      secret: {true|false} # Specifies if the property is confidential
                    }
                  ]
                }
              ]
              configuration: [
                { 
                  key: "{CONFIG_PROPERTY_KEY}"
                  value: "{CONFIG_PROPERTY_VALUE}"
                  secret: {true|false} # Specifies if the property is confidential
                }
              ]
            }

>             credentials: { secretName: "{GARDENER_SERVICE_ACCOUNT_CONFIGURATION_SECERT_NAME}" }

Reviewer 2: Please remove

  4. Download the service account configuration (`kubeconfig.yaml`) and use it to create a Secret in the `compass-system` Namespace with the key `credentials` and the value encoded with base64.

     ```bash
     kubectl -n compass-system create secret generic {SECRET_NAME} --from-file=credentials={PATH_TO_KUBECONFIG}
     ```

>   5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on AWS. Note, that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory, and that the order of the components matters.

Reviewer: ```suggestion
  5. Make a call to the Runtime Provisioner with a **tenant** header to create a cluster on AWS. Note that the Runtime Agent component (`compass-runtime-agent`) in the Kyma configuration is mandatory and that the order of the components matters.
```

                  awsConfig: {
                    publicCidr: "10.250.96.0/22"
                    vpcCidr: "10.250.0.0/16"
                    internalCidr: "10.250.112.0/22"
                    zone: "eu-west-1b"
                  } 
                }
              }
            }
            kymaConfig: {
              version: "1.8.0"
              components: [
                { component: "compass-runtime-agent", namespace: "compass-system" }
                {
                  component: "{KYMA_COMPONENT_NAME}"
                  namespace: "{NAMESPACE_TO_INSTALL_COMPONENT_TO}"
                  configuration: [
                    { key: "{CONFIG_PROPERTY_KEY}"
                      value: "{CONFIG_PROPERTY_VALUE}"
                      secret: {true|false} # Specifies if the property is confidential
                    }
                  ]
                }
              ]
              configuration: [
                { 
                  key: "{CONFIG_PROPERTY_KEY}"
                  value: "{CONFIG_PROPERTY_VALUE}"
                  secret: {true|false} # Specifies if the property is confidential
                }
              ]
            }

>             credentials: { secretName: "{GARDENER_SERVICE_ACCOUNT_CONFIGURATION_SECERT_NAME}" }

Reviewer 2: Please remove this from the example call

      ```graphql
      {
        "data": {
          "provisionRuntime": {
            "runtimeID": "{RUNTIME_ID}",
            "id": "{OPERATION_ID}"
          }
        }
      }
      ```
  </details>
    
</div>

> The operation of provisioning is asynchronous. The operation of provisioning returns the Runtime Operation Status containing the runtime ID (`provisionRuntime.runtimeID`) and the operation ID (`provisionRuntime.id`). Use the Runtime ID to [check the Runtime Status](08-04-runtime-status.md). Use the provisioning operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md) and verify that the provisioning was successful.

Reviewer: ```suggestion
The operation of provisioning is asynchronous. Provisioning returns the Runtime Operation Status that contains the Runtime ID (`provisionRuntime.runtimeID`) and the operation ID (`provisionRuntime.id`). Use the Runtime ID to [check the Runtime Status](08-04-runtime-status.md). Use the provisioning operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md) and verify that the provisioning is successful.
```
